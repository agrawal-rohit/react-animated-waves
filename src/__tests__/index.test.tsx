import React from 'react';
import { render } from '@testing-library/react';
import AudioWaves from '../index';
import '@testing-library/jest-dom';

// Mock canvas context
HTMLCanvasElement.prototype.getContext = (() => {
  // return whatever context you want to use in your tests
  return {
    canvas: {
      width: 500,
      height: 500,
      style: {
        width: '100%',
        height: '100%',
      },
    },
    fillRect: function () {},
    clearRect: function () {},
    getImageData: function (x: any, y: any, w: any, h: any) {
      return {
        data: new Array(w * h * 4),
      };
    },
    putImageData: function () {},
    createImageData: function () {
      return [];
    },
    setTransform: function () {},
    drawImage: function () {},
    save: function () {},
    fillText: function () {},
    restore: function () {},
    beginPath: function () {},
    moveTo: function () {},
    lineTo: function () {},
    closePath: function () {},
    stroke: function () {},
    translate: function () {},
    scale: function () {},
    rotate: function () {},
    arc: function () {},
    fill: function () {},
    measureText: function () {
      return { width: 0 };
    },
    transform: function () {},
    rect: function () {},
    clip: function () {},
    createLinearGradient: function () {
      return {
        addColorStop: function () {},
      };
    },
  };
}) as any;

describe('AudioWaves Component', () => {
  it('renders without crashing', () => {
    const { container } = render(<AudioWaves />);
    expect(container).toMatchSnapshot();
  });

  it('renders with default props', () => {
    const { container } = render(<AudioWaves />);
    expect(container).toMatchSnapshot();
  });

  it('renders with custom amplitude', () => {
    const { container } = render(<AudioWaves amplitude={50} />);
    expect(container).toMatchSnapshot();
  });

  it('renders with custom colors', () => {
    const { container } = render(
      <AudioWaves colors={['#FF0000', '#00FF00', '#0000FF']} />
    );
    expect(container).toMatchSnapshot();
  });
});
