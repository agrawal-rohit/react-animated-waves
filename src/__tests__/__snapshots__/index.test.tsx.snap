// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioWaves Component renders with custom amplitude 1`] = `
<div>
  <canvas
    height="auto"
    width="0"
  />
</div>
`;

exports[`AudioWaves Component renders with custom colors 1`] = `
<div>
  <canvas
    height="auto"
    width="0"
  />
</div>
`;

exports[`AudioWaves Component renders with default props 1`] = `
<div>
  <canvas
    height="auto"
    width="0"
  />
</div>
`;

exports[`AudioWaves Component renders without crashing 1`] = `
<div>
  <canvas
    height="auto"
    width="0"
  />
</div>
`;
